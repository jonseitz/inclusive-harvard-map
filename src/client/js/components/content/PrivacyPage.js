import React from 'react';
import ReactMarkdown from 'react-markdown';

/**
 * Renders a message about privacy while using the app, including an explanation about location data.
 * @extends React.Component
 */

const content = `
This application does not store any personal identifying information about its users, and wherever possible limits the sharing of that information with groups or corporations that may do so.

By default, location data from your device is not used. If you choose to use your location data, your coordinates may be stored in server logs along with the public IP address of the device making the request. It will not be shared with any outside vendors or services.

**The privacy of your data matters. You should haven't to trade that privacy away just to find a restroom.**
`;

class PrivacyPage extends React.Component {
  render() {
    return (
      <ReactMarkdown>
        {content}
      </ReactMarkdown>);
  }
}

export default PrivacyPage;
