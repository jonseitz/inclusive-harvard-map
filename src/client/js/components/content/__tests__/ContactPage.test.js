import React from 'react';
import enzyme from 'enzyme';
import PrivacyPage from '../PrivacyPage';

describe('PrivacyPage', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = enzyme.shallow(<PrivacyPage />);
  });
  it('Should render the content without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
