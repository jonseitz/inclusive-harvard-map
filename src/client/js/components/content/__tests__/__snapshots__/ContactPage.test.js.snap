// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyPage Should render the content without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrivacyPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "astPlugins": Array [],
      "children": "
This application does not store any personal identifying information about its users, and wherever possible limits the sharing of that information with groups or corporations that may do so.

By default, location data from your device is not used. If you choose to use your location data, your coordinates may be stored in server logs along with the public IP address of the device making the request. It will not be shared with any outside vendors or services.

**The privacy of your data matters. You should haven't to trade that privacy away just to find a restroom.**
",
      "escapeHtml": true,
      "plugins": Array [],
      "rawSourcePos": false,
      "renderers": Object {},
      "skipHtml": false,
      "sourcePos": false,
      "transformLinkUri": [Function],
    },
    "ref": null,
    "rendered": "
This application does not store any personal identifying information about its users, and wherever possible limits the sharing of that information with groups or corporations that may do so.

By default, location data from your device is not used. If you choose to use your location data, your coordinates may be stored in server logs along with the public IP address of the device making the request. It will not be shared with any outside vendors or services.

**The privacy of your data matters. You should haven't to trade that privacy away just to find a restroom.**
",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "astPlugins": Array [],
        "children": "
This application does not store any personal identifying information about its users, and wherever possible limits the sharing of that information with groups or corporations that may do so.

By default, location data from your device is not used. If you choose to use your location data, your coordinates may be stored in server logs along with the public IP address of the device making the request. It will not be shared with any outside vendors or services.

**The privacy of your data matters. You should haven't to trade that privacy away just to find a restroom.**
",
        "escapeHtml": true,
        "plugins": Array [],
        "rawSourcePos": false,
        "renderers": Object {},
        "skipHtml": false,
        "sourcePos": false,
        "transformLinkUri": [Function],
      },
      "ref": null,
      "rendered": "
This application does not store any personal identifying information about its users, and wherever possible limits the sharing of that information with groups or corporations that may do so.

By default, location data from your device is not used. If you choose to use your location data, your coordinates may be stored in server logs along with the public IP address of the device making the request. It will not be shared with any outside vendors or services.

**The privacy of your data matters. You should haven't to trade that privacy away just to find a restroom.**
",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
